NRLMMD-GEOIPS/geoips
# Provide better error message for --compare_path without testing dependencies

# Related Issues
fixes NRLMMD-GEOIPS/geoips#451

# Testing Instructions
* Enter a `bash` environment in the Dockerfile, remove `pixelmatch`, run `tests/integration_tests/base_install.sh` and then `tests/integration_tests/base_test.sh`. `cat /output/logs/20240408/20240408.223821_geoips_base/test_all_geoips_base.log_amsr2_ocean.tc.windspeed.imagery_clean.sh.log` to see error message:

```bash
Processing complete! Checking outputs...


Traceback (most recent call last):
  File "/app/geoips_packages/geoips/geoips/plugins/modules/output_checkers/image.py", line 123, in outputs_match
    from pixelmatch.contrib.PIL import pixelmatch
ModuleNotFoundError: No module named 'pixelmatch'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/geoips_user/.local/bin/run_procflow", line 8, in <module>
    sys.exit(main())
  File "/app/geoips_packages/geoips/geoips/commandline/run_procflow.py", line 69, in main
    RETVAL = PROCFLOW(COMMAND_LINE_ARGS["filenames"], COMMAND_LINE_ARGS)
  File "/app/geoips_packages/geoips/geoips/plugins/modules/procflows/single_source.py", line 1917, in call
    retval += output_checker(
  File "/app/geoips_packages/geoips/geoips/plugins/modules/output_checkers/image.py", line 230, in call
    retval = plugin.compare_outputs(
  File "/app/geoips_packages/geoips/geoips/interfaces/module_based/output_checkers.py", line 849, in compare_outputs
    ) = self.perform_comparisons(
  File "/app/geoips_packages/geoips/geoips/interfaces/module_based/output_checkers.py", line 776, in perform_comparisons
    goodcomps, badcomps, compare_strings = test_product_func(
  File "/app/geoips_packages/geoips/geoips/interfaces/module_based/output_checkers.py", line 965, in test_products
    if self.module.outputs_match(
  File "/app/geoips_packages/geoips/geoips/plugins/modules/output_checkers/image.py", line 125, in outputs_match
    raise ImportError(
ImportError: Comparing images using GeoIPS requires a python library called pixelmatch (link: https://pypi.org/project/pixelmatch/).You can install it directly or install the test extra eg. pip install geoips[test]
```

# Summary
* Added a clear and informative error message when `--compare_path` is used without the pixelmatch dependency installed.
* Checked for other testing functionalities in `run_procflow` and added appropriate error messages for missing dependencies. See #451 for details.

# Output
* Command line output demonstrating the new error message when `--compare_path` is used without installing pixelmatch.
* No imagery output is required as the changes are related to error handling.
